shell/platform/fuchsia/flutter/software_surface.cc

```diff
@@ -0,0 +1,404 @@
+    args.set_buffer_collection_token(std::move(scenic_token));
+    args.set_usage(
+        fuchsia::ui::composition::RegisterBufferCollectionUsage::DEFAULT);
+    flatland_allocator->RegisterBufferCollection(
+        std::move(args),
+        [](fuchsia::ui::composition::Allocator_RegisterBufferCollection_Result
+               result) {
+          if (result.is_err()) {
+            FML_LOG(ERROR)
+                << "RegisterBufferCollection call to Scenic Allocator failed";
+          }
+        });
+  }
+
+  fuchsia::sysmem::BufferCollectionSyncPtr buffer_collection;
+  zx_status_t bind_status = sysmem_allocator->BindSharedCollection(
+      std::move(local_token), buffer_collection.NewRequest());
+  if (bind_status != ZX_OK) {
+    FML_LOG(ERROR) << "SoftwareSurface: Failed to bind collection token: "
+                   << zx_status_get_string(bind_status);
+    return false;
+  }
+
+  fuchsia::sysmem::BufferCollectionConstraints constraints;
+  constraints.min_buffer_count = 1;
+  constraints.usage.cpu =
+      fuchsia::sysmem::cpuUsageWrite | fuchsia::sysmem::cpuUsageWriteOften;
+  constraints.has_buffer_memory_constraints = true;
+  constraints.buffer_memory_constraints.min_size_bytes = 0;
+  constraints.buffer_memory_constraints.max_size_bytes = 0xffffffff;
+  constraints.buffer_memory_constraints.physically_contiguous_required = false;
+  constraints.buffer_memory_constraints.secure_required = false;
+  constraints.buffer_memory_constraints.ram_domain_supported = true;
+  constraints.buffer_memory_constraints.cpu_domain_supported = true;
+  constraints.buffer_memory_constraints.inaccessible_domain_supported = false;
+  constraints.buffer_memory_constraints.heap_permitted_count = 0;
+  constraints.image_format_constraints_count = 1;
+  fuchsia::sysmem::ImageFormatConstraints& image_constraints =
+      constraints.image_format_constraints[0];
+  image_constraints = fuchsia::sysmem::ImageFormatConstraints();
+  image_constraints.min_coded_width = static_cast<uint32_t>(size.fWidth);
+  image_constraints.min_coded_height = static_cast<uint32_t>(size.fHeight);
+  image_constraints.max_coded_width = static_cast<uint32_t>(size.fWidth);
+  image_constraints.max_coded_height = static_cast<uint32_t>(size.fHeight);
+  image_constraints.min_bytes_per_row = 0;
+  image_constraints.max_bytes_per_row = std::numeric_limits<uint32_t>::max();
+  image_constraints.max_coded_width_times_coded_height =
+      std::numeric_limits<uint32_t>::max();
+  image_constraints.pixel_format.type =
+      fuchsia::sysmem::PixelFormatType::R8G8B8A8;
+  image_constraints.color_spaces_count = 1;
+  image_constraints.color_space[0].type = fuchsia::sysmem::ColorSpaceType::SRGB;
+  image_constraints.pixel_format.has_format_modifier = true;
+  image_constraints.pixel_format.format_modifier.value =
+      fuchsia::sysmem::FORMAT_MODIFIER_LINEAR;
+  image_constraints.min_bytes_per_row = static_cast<uint32_t>(size.fWidth) * 4;
```

Comment:
These should be implied when you FORMAT_MODIFIER_LINEAR, but no harm in specifying them again.